server/channels/app/plugin_api_tests/test_update_user_auth_plugin/main.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+	"github.com/mattermost/mattermost/server/v8/channels/app/plugin_api_tests"
+)
+
+type MyPlugin struct {
+	plugin.MattermostPlugin
+	configuration plugin_api_tests.BasicConfig
+}
+
+func (p *MyPlugin) OnConfigurationChange() error {
+	if err := p.API.LoadPluginConfiguration(&p.configuration); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *MyPlugin) expectUserAuth(userID string, expectedUserAuth *model.UserAuth) error {
+	user, err := p.API.GetUser(p.configuration.BasicUserID)
+	if err != nil {
+		return err
+	}
+	if user.AuthService != expectedUserAuth.AuthService {
+		return fmt.Errorf("expected '%s' got '%s'", expectedUserAuth.AuthService, user.AuthService)
+	}
+	if user.AuthData == nil && expectedUserAuth.AuthData != nil {
+		return fmt.Errorf("expected '%s' got nil", *expectedUserAuth.AuthData)
+	} else if user.AuthData != nil && expectedUserAuth.AuthData == nil {
+		return fmt.Errorf("expected nil got '%s'", *user.AuthData)
+	} else if user.AuthData != nil && expectedUserAuth.AuthData != nil && *user.AuthData != *expectedUserAuth.AuthData {
+		return fmt.Errorf("expected '%s' got '%s'", *expectedUserAuth.AuthData, *user.AuthData)
```

Comment:
Would it be easier to do a deepEqual of some kind here?

Reply:
Nothing specific, I would just use the one in `testify.Assert.DeepEqual` I think? But anyway not crucial, just thought it would be easier. :)

Reply:
Oh, I didn't realize it was simply exported! Realistically, I probably won't touch this PR, but save myself the effort next time and fix this then :)